MySQL source documentation (#617)
rudderlabs/rudderstack-docs: docs/reverse-etl/mysql.mdx

---
# slug: "/docs/reverse-etl/postgresql"
title: "MySQL"
description: Step-by-step guide to ingest your data from MySQL into RudderStack.
---

# MySQL

> [MySQL](https://www.mysql.com/) is an open source relational DBMS. It provides services to run your OLTP, OLAP, and machine learning workloads directly on your database with high scalability and reliability.

Author: ```suggestion
[MySQL](https://www.mysql.com/) is a popular open source relational DBMS.
```

RudderStack supports MySQL as a source from which you can ingest data and route it to your desired downstream destinations.

## Granting permissions

RudderStack requires you to grant certain user permissions on your MySQL database to successfully access data from it.

Run the SQL queries listed in the following sections in the  **exact order** to grant these permissions:

### Step 1: Creating the user

- Create a new user `RUDDER` with a password `<strong_unique_password>` in MySQL, as shown:

```sql

> CREATE USER RUDDER IDENTIFIED BY 'password';

Reviewer: can we change this to '<password>'

Reviewer: as discussed

```

<div class="warningBlock">

Set a strong, unique password for creating the user in MySQL.
</div>

### Step 2: Creating the RudderStack schema and granting permissions

1. Create a dedicated schema `_rudderstack`.

```sql

> CREATE SCHEMA "_rudderstack";

Reviewer: this will be create schema _rudderstack; 
Please remove the double quotes

2. From the left navigation bar, go to **Source** > **New Source** > **Reverse ETL**.  Then, select **MySQL**.
3. Assign a name to your source and click on **Continue**.

### Configuring the connection credentials

1. Enter the relevant connection details in the **Connection Credentials**, as shown:

<img src="../assets/warehouse-actions-sources/mysql-connection-settings.png" />

* **Host**: Enter the host name of your MySQL service.
* **Port**: Enter the port number of your MySQL service.
* **Database**: Enter the database name in your MySQL instance where the data is loaded.
* **User**: Specify the username which has the required read/write access to the above database.
* **Password**: Enter the password for the above user.

<div class="infoBlock">

If you've configured a MySQL source before, you can select the existing credentials under the <strong>Use existing credentials</strong> option.
</div>

3. Click on **Continue**. RudderStack will then verify and validate your credentials. 

<div class="infoBlock">

For more information on these validation steps, refer to the <a href="#faq">FAQ</a> section. 
</div>

> 4. Once verified, click on **Continue** to proceed.

Reviewer 2: ```suggestion
3. Once verified, click on **Continue** to proceed.
```
